webhook/resources/recurringjob/mutator.go

```diff
@@ -51,6 +53,23 @@ func (r *recurringJobMutator) Create(request *admission.Request, newObj runtime.
 		patchOps = append(patchOps, `{"op": "replace", "path": "/spec/labels", "value": {}}`)
 	}
 
+	log := logrus.WithFields(logrus.Fields{
+		"recurringJob": recurringjob.Name,
+		"task":         recurringjob.Spec.Task,
+	})
+	switch recurringjob.Spec.Task {
+	case longhorn.RecurringJobTypeSnapshotCleanup:
+		if recurringjob.Spec.Retain != 0 {
+			log.Debugf("Replacing ineffective retain value in RecurringJob: from %v to 0", recurringjob.Spec.Retain)
+			patchOps = append(patchOps, `{"op": "replace", "path": "/spec/retain", "value": 0}`)
+		}
+	default:
+		if recurringjob.Spec.Retain < 1 {
```

Comment:
For task snapshot delete, can we set retain to 0?

Reply:
Do you mean making retain value ineffective for task snapshot delete?

Reply:
As I mentioned above, 0 means retaining the volume head only.

Reply:
Got it! Thank you and updated.